c-dependencies/js-compute-runtime/xqd.wit

```diff
@@ -0,0 +1,510 @@
+      *
+      * These are available for some IP addresses in Australia, Canada, France, Germany, Italy, Spain, Switzerland, the United Kingdom, and the United States.
+      *
+      * For Canadian postal codes, this is the first 3 characters. For the United Kingdom, this is the first 2-4 characters (outward code). For countries with alphanumeric postal codes, this field is a lowercase transliteration.
+      */
+      postal-code: option<string>,
+
+      /**
+      * Client proxy description.
+      */
+      proxy-description: option<string>,
+
+      /**
+      * Client proxy type.
+      */
+      proxy-type: option<string>,
+
+      /**
+      * [ISO 3166-2](https://en.wikipedia.org/wiki/ISO_3166-2) country subdivision code.
+      *
+      * For countries with multiple levels of subdivision (for example, nations within the United Kingdom), this variable gives the more specific subdivision.
+      *
+      * This field can be None for countries that do not have ISO country subdivision codes. For example, None is given for IP addresses assigned to the Åland Islands (country code AX, illustrated below).
+      */
+      region: option<string>,
+
+      /**
+      * Time zone offset from coordinated universal time (UTC) for `city`.
+      */
+      utc-offset: option<u32>,
+    }
+
+    /// JSON string for now
+    geo-lookup: func(addr-octets: list<u8>) -> result<string, error>
+
+
+    /*
+     * Fastly KV
+     */
+    type kv-store-handle = u32
+
+    kv-open: func(name: string) -> result<kv-store-handle, error>
+
+    kv-lookup: func(store: kv-store-handle, key: list<u8>) -> result<option<body-handle>, error>
+
+    kv-insert: func(store: kv-store-handle, key: list<u8>, body-handle: body-handle, max-age: u32) -> result<bool, error>
+
+
+    /*
+     * Fastly Object Store
+     */
+    type fd = u32
+    type object-store-handle = u32
+
+    object-store-open: func(name: string) -> result<object-store-handle, error>
+
+    object-store-lookup: func(store: object-store-handle, key: string) -> result<option<body-handle>, error>
+
+    object-store-lookup-as-fd: func(store: object-store-handle, key: string) -> result<option<fd>, error>
+
+    // Should object store insert return "inserted" bool?
+    object-store-insert: func(store: object-store-handle, key: string, body-handle: body-handle) -> result<_, error>
+
+
+    /*
+     * Fastly Secret Store
+     */
+    type secret-store-handle = u32
+    type secret-handle = u32
+
+    secret-store-open: func(name: string) -> result<secret-store-handle, error>
+
+    secret-store-get: func(store: secret-store-handle, key: string) -> result<option<secret-handle>, error>
+
+    secret-store-plaintext: func(secret: secret-handle) -> result<option<string>, error>
+
+
+    /*
+     * Fastly Backend
+     */
+    enum backend-health {
+      unknown,
+      healthy,
+      unhealthy,
+    }
+
+    backend-is-healthy: func(backend: string) -> result<backend-health, error>
```

Comment:
This doesn't have a public ABI definition as of today (It is not in the public wit stored in Viceroy) - which is the deciding factor on whether we include a feature in the public SDKs. Do you know if that deciding factor changes now that we use the component model?